### PR TITLE
chore(RHINENG-17921): Update delete duplicates job

### PR DESCRIPTION
**Pull request by fstavela**
# Overview

This PR is being created to address [RHINENG-17921](https://issues.redhat.com/browse/RHINENG-17921).

This PR updates the duplicate-hosts-remover job to use the same logic as the current deduplication process uses. We will need to run this job once to remove the duplicates that we created during the [MQ batching issues](https://issues.redhat.com/browse/RHINENG-13683).

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by kruai — DISMISSED** [review]
I originally had a couple of concerns about ordering, which duplicates got deleted, etc... but the ordering is actually fine, and I expect this to work as designed. I imagine the main potential issue is that this may take a long time to run because of the number of queries it makes; however, I can't see a way to remedy that atm. LGTM

File: lib/host_remove_duplicates.py [hunk 1]
```diff
@@ -1,93 +1,62 @@
...
-    return query
-
 
-def contains_no_elevated_facts_query(query):
-    return query.filter(not_(Host.canonical_facts.has_any(array(ELEVATED_CANONICAL_FACT_FIELDS))))
+def find_matching_hosts(canonical_facts: dict[str, Any], query: Query) -> list[Host]:
+    immutable_facts, id_facts = extract_immutable_and_id_facts(canonical_facts)
 
+    # First search based on immutable ID facts.
+    if immutable_facts:  # noqa: SIM102
```
Comment (thearifismail) [inline lib/host_remove_duplicates.py:22]:
```suggestion
    if immutable_facts and existing_hosts := find_hosts_by_multiple_facts(immutable_facts, query):
```

Comment (thearifismail) [reply lib/host_remove_duplicates.py:22]: Safe to combine because there is no `else`

**Review by thearifismail — DISMISSED** [review]
LGTM. A minor comment but not worth stopping the merge.
